libs/common/src/archive.rs

```diff
@@ -35,3 +36,40 @@ pub fn archive<P: AsRef<Path>>(folder: P) -> Result<Option<String>> {
         Ok(Some(STANDARD.encode(bytes)))
     }
 }
+
+pub fn archive_entries(dir_walker: Walk, suffix: Option<&Path>) -> Result<Option<String>> {
+    let encoder = GzEncoder::new(Vec::new(), Compression::default());
+    let mut tar = tar::Builder::new(encoder);
+    let mut count = 0;
+    for result in dir_walker {
+        match result {
+            Ok(entry) => {
+                let path = entry.path();
+                if suffix.is_some() && !path.ends_with(suffix.unwrap()) {
+                    continue;
+                }
+                // Note: tar automatically removes the common prefix
+                // a/b/c/a.ts, a/b, a/b/d.ts => c/a.ts, .,  d.ts
+                if path.is_dir() {
+                    tar.append_dir_all(".", path)
+                        .context("Adding directory to tarball")?;
+                } else {
+                    let mut file = fs::File::open(path)
+                        .context(format!("failed to open file {}", path.display()))?;
+                    tar.append_file(".", &mut file)
+                        .context("Adding file to tarball")?;
+                }
+                count += 1;
+                Ok(())
+            }
+            Err(e) => Err(e),
+        }?;
+    }
+
+    if count > 0 {
```

Comment:
why is the count required? this is not supported by the tar?

Reply:
It does not have is_empty, count or similar method. Can only add files/folder.